src/SAML2/XML/saml/AuthnContextDecl.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+declare(strict_types=1);
+
+namespace SAML2\XML\saml;
+
+use DOMElement;
+use DOMNodeList;
+use SAML2\Constants;
+use SAML2\DOMDocumentFactory;
+use SAML2\Utils;
+use Webmozart\Assert\Assert;
+
+/**
+ * Class representing SAML2 AuthnContextDecl
+ *
+ * @author Tim van Dijen, <tvdijen@gmail.com>
+ * @package SimpleSAMLphp
+ */
+class AuthnContextDecl extends \SAML2\XML\AbstractConvertable
+{
+    /** @var \DOMNodeList */
+    private $decl;
+
+
+    /**
+     * Initialize an AuthnContextDecl.
+     *
+     * @param \DOMNodeList $decl
+     */
+    public function __construct(DOMNodeList $decl)
+    {
+        $this->setDecl($decl);
+    }
+
+
+    /**
+     * Collect the value of the decl-property
+     *
+     * @return \DOMNodeList
+     *
+     * @throws \InvalidArgumentException if assertions are false
+     */
+    public function getDecl(): DOMNodeList
+    {
+        return $this->decl;
+    }
+
+
+    /**
+     * Set the value of the decl-property
+     *
+     * @param \DOMNodeList $decl
+     * @return void
+     */
+    private function setDecl(DOMNodeList $decl): void
+    {
+        $this->decl = $decl;
+    }
+
+
+    /**
+     * Un-serialize this XML chunk.
+     *
+     * @param string $serialized The serialized chunk.
+     * @return void
+     *
+     * Type hint not possible due to upstream method signature
+     */
+    public function unserialize($serialized): void
```

Comment:
This shouldn't be necessary, I believe. The parent calling `fromXML()` should be enough to recreate the object, right?